drivers/adc/adc_bflb.c

```diff
@@ -0,0 +1,700 @@
+/*
+ * Copyright (c) 2024-2025 MASSDRIVER EI (massdriver.space)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT bflb_adc
+
+#include <zephyr/irq.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/syscon.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(adc_bflb, CONFIG_ADC_LOG_LEVEL);
+
+#include <bflb_soc.h>
+#include <glb_reg.h>
+#include <hbn_reg.h>
+#include <bouffalolab/common/adc_reg.h>
+#include <zephyr/drivers/clock_control/clock_control_bflb_common.h>
+
```

Comment:
If interested in ADC `lock()`/`unlock()` support and related:

```suggestion
#define ADC_CONTEXT_USES_KERNEL_TIMER
#include "adc_context.h"
```

https://github.com/zephyrproject-rtos/zephyr/blob/db0c34d836aeb79ddbdff0b219499009eebc59d9/drivers/adc/adc_context.h#L33-L36

Reply:
I thought so. thanks for checking